index.js

```diff
@@ -15,6 +15,44 @@ const getFilenameFromMime = (name, mime) => {
 	return `${name}.${extensions[0].ext}`;
 };
 
+const majorElectronVersion = () => {
+	const version = process.versions.electron.split('.');
+	return parseInt(version[0], 10);
+};
+
+const getWindowFromBrowserView = webContents => {
+	for (const currentWindow of BrowserWindow.getAllWindows()) {
+		for (const currentBrowserView of currentWindow.getBrowserViews()) {
+			if (currentBrowserView.webContents.id === webContents.id) {
+				return currentWindow;
+			}
+		}
+	}
+};
+
+const getWindowFromWebcontents = webContents => {
+	let window_;
+	const webContentsType = webContents.getType();
+	switch (webContentsType) {
+		case 'webview':
+			window_ = BrowserWindow.fromWebContents(webContents.hostWebContents);
+			break;
+		case 'browserView':
+			if (majorElectronVersion() < 12) {
+				window_ = getWindowFromBrowserView(webContents);
+			} else {
+				window_ = BrowserWindow.fromWebContents(webContents);
```

Comment:
Shouldn't this use `.fromBrowserView`?

Reply:
no, we don't have the browserview at that line, only the webcontents. The answer in the electron ticket fixed being able to get the window from a webcontents inside a browserview. I really don't need the browserview, but considering I'm not going to be the only user of that I asked for the one that most people would use, but the answer was that getting the window from the nested webcontent was fixed and that's ok for our current usecase.

Reply:
Ah ok. Thanks for elaborating. It seems the mentioned fix will handle nested webviews too, so I think in the future this whole method could be removed.

Could you instead use this one line 73?

```js
let window_;
if (majorElectronVersion() >= 12) {
	window_ = BrowserWindow.fromWebContents(webContents);
} else {
	window_ = getWindowFromWebcontents(webContents);
}
```